Source/Renderer/Buffer.js

```diff
@@ -23,6 +57,111 @@ define([
+     * of the index buffer defined by an offset and count.
+     *
+     * @param {Context} context The context in which to create the buffer
+     * @param {ArrayBufferView|Number} typedArrayOrSizeInBytes A typed array containing the data to copy to the buffer, or a <code>Number</code> defining the size of the buffer in bytes.
+     * @param {BufferUsage} usage Specifies the expected usage pattern of the buffer.  On some GL implementations, this can significantly affect performance.  See {@link BufferUsage}.
+     * @param {IndexDatatype} indexDatatype The datatype of indices in the buffer.
+     * @returns {IndexBuffer} The index buffer, ready to be attached to a vertex array.
+     *
+     * @exception {DeveloperError} IndexDatatype.UNSIGNED_INT requires OES_element_index_uint, which is not supported on this system.    Check context.elementIndexUint.
+     * @exception {DeveloperError} The size in bytes must be greater than zero.
+     * @exception {DeveloperError} Invalid <code>usage</code>.
+     * @exception {DeveloperError} Invalid <code>indexDatatype</code>.
+     *
+     * @see Buffer#createVertexBuffer
+     * @see Context#draw
+     * @see {@link https://www.khronos.org/opengles/sdk/docs/man/xhtml/glGenBuffer.xml|glGenBuffer}
+     * @see {@link https://www.khronos.org/opengles/sdk/docs/man/xhtml/glBindBuffer.xml|glBindBuffer} with <code>ELEMENT_ARRAY_BUFFER</code>
+     * @see {@link https://www.khronos.org/opengles/sdk/docs/man/xhtml/glBufferData.xml|glBufferData} with <code>ELEMENT_ARRAY_BUFFER</code>
+     *
+     * @example
+     * // Example 1. Create a stream index buffer of unsigned shorts that is
+     * // 16 bytes in size.
+     * var buffer = Buffer.createIndexBuffer(context, 16, BufferUsage.STREAM_DRAW,
+     *     IndexDatatype.UNSIGNED_SHORT);
+     *
+     * @example
+     * // Example 2. Create a static index buffer containing three unsigned shorts.
+     * var buffer = Buffer.createIndexBuffer(context, new Uint16Array([0, 1, 2]),
+     *     BufferUsage.STATIC_DRAW, IndexDatatype.UNSIGNED_SHORT)
+     */
+    Buffer.createIndexBuffer = function(context, typedArrayOrSizeInBytes, usage, indexDatatype) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!IndexDatatype.validate(indexDatatype)) {
+            throw new DeveloperError('Invalid indexDatatype.');
+        }
+        //>>includeEnd('debug');
+
+        if ((indexDatatype === IndexDatatype.UNSIGNED_INT) && !context.elementIndexUint) {
+            throw new DeveloperError('IndexDatatype.UNSIGNED_INT requires OES_element_index_uint, which is not supported on this system.  Check context.elementIndexUint.');
+        }
+
+        var bytesPerIndex = IndexDatatype.getSizeInBytes(indexDatatype);
+        var buffer = new Buffer(context, context._gl.ELEMENT_ARRAY_BUFFER, typedArrayOrSizeInBytes, usage);
+        var numberOfIndices = buffer.sizeInBytes / bytesPerIndex;
+
+        defineProperties(buffer, {
```

Comment:
You're sure this appends properties and that it doesn't overwrite the existing ones?